### ISSUE TITLE
Recover from failed broadcasts -  was not in database and not on network

### ISSUE THREAD

**Issue by tzarebczan** [lbryio/lbry-sdk#2462]
I recently ran into an issue where my wallet thought my transaction was not broadcast, so it didn't appear in channel list / transaction list until after it was confirmed (this prevented me from publishing until confirmed). I've also run into a similar issue where the transaction was not broadcast at all and effectively lost.  

In either case, we should handle this scenario better. The SDK should know that it was not broadcast/retrieved correctly and re-try retrieving/re-broadcasting as appropriate. 

```
Task exception was never retrieved
future: <Task finished coro=<BaseLedger.update_history() done, defined at torba\client\baseledger.py:415> exception=ValueError('Transaction 58802477358348fa833e592706a54c89c79a17a51270e7f0e39249b82f864d8d was not in database and not on network.') created at torba\tasks.py:12>
source_traceback: Object created at (most recent call last):
  File "lbry\extras\cli.py", line 317, in <module>
  File "lbry\extras\cli.py", line 294, in main
  File "lbry\extras\cli.py", line 269, in run_daemon
  File "asyncio\base_events.py", line 534, in run_forever
  File "asyncio\base_events.py", line 1763, in _run_once
  File "asyncio\events.py", line 88, in _run
  File "torba\rpc\session.py", line 448, in _throttled_request
  File "torba\client\basenetwork.py", line 132, in handle_request
  File "torba\stream.py", line 84, in add
  File "torba\stream.py", line 70, in _notify_and_ensure_future
  File "torba\stream.py", line 84, in <lambda>
  File "torba\stream.py", line 32, in _add
  File "torba\client\baseledger.py", line 413, in process_status_update
  File "torba\tasks.py", line 12, in add
Traceback (most recent call last):
  File "torba\client\baseledger.py", line 445, in update_history
  File "torba\client\baseledger.py", line 529, in cache_transaction
ValueError: Transaction 58802477358348fa833e592706a54c89c79a17a51270e7f0e39249b82f864d8d was not in database and not on network.
2019-09-16 16:05:27,603 INFO     torba.client.baseledger:120: (lbc_mainnet) on_transaction: address=bRouEChU3DGjcnDm99u3dMedLkw1cS7ThT, height=-1, is_verified=False, tx.id=58802477358348fa833e592706a54c89c79a17a51270e7f0e39249b82f864d8d
2019-09-16 16:05:27,611 INFO     torba.client.baseledger:120: (lbc_mainnet) on_transaction: address=bNWMyu9W16HxokfzxpZL87BL8ifsGMZL7B, height=-1, is_verified=False, tx.id=58802477358348fa833e592706a54c89c79a17a51270e7f0e39249b82f864d8d
```

Later found: 
```2019-09-16 16:08:43,177 INFO     torba.client.baseledger:135: lbc_mainnet: added BlockHeightEvent(height=634724, change=1) header blocks, final height 634724
2019-09-16 16:08:43,227 DEBUG    lbry.dht.protocol.protocol:424: 24.184.108.248:4444 RECV CALL findNode 67.163.78.116:4444
2019-09-16 16:08:43,977 INFO     torba.client.baseledger:120: (lbc_mainnet) on_transaction: address=bU2Hh66LqUfGXhij7nJ2qgapsUaH83v5UF, height=634724, is_verified=True, tx.id=58802477358348fa833e592706a54c89c79a17a51270e7f0e39249b82f864d8d
2019-09-16 16:08:44,023 INFO     torba.client.baseledger:120: (lbc_mainnet) on_transaction: address=bNWMyu9W16HxokfzxpZL87BL8ifsGMZL7B, height=634724, is_verified=True, tx.id=58802477358348fa833e592706a54c89c79a17a51270e7f0e39249b82f864d8d
2019-09-16 16:08:44,045 INFO     torba.client.baseledger:127: (lbc_mainnet) on_address: ['bJCiZZVft8DwSzuAQUVH4ChTMDgnfrQ3Q8', 'bXVNmwbt9wtNUuK6kpGkWxYXZQNn48ihBn', 'bQP7UzeUEefii5CHUL6WZeLZAAzAouSaeq']
2019-09-16 16:08:44,071 INFO     torba.client.baseledger:120: (lbc_mainnet) on_transaction: address=bRouEChU3DGjcnDm99u3dMedLkw1cS7ThT, height=634724, is_verified=True, tx.id=b208c603419146ce7bdb16ded569c399755e9f8baa57bd6b2eba962d0702f00e
2019-09-16 16:08:44,072 INFO     torba.client.baseledger:120: (lbc_mainnet) on_transaction: address=bRouEChU3DGjcnDm99u3dMedLkw1cS7ThT, height=634724, is_verified=True, tx.id=58802477358348fa833e592706a54c89c79a17a51270e7f0e39249b82f864d8d```

**Comment by tzarebczan** [comment]
If using --blocking, the call does not return at all until the next block hits. We should still support rebroadcasting these...it's a poor user experience otherwise.

```019-10-04 11:12:48,211 INFO     torba.client.baseledger:136: lbc_mainnet: added BlockHeightEvent(height=644355, change=1) header blocks, final height 644355
2019-10-04 11:17:34,756 INFO     torba.client.baseledger:121: (lbc_mainnet) on_transaction: address=bPqf8FWmDrPiygFb9KWWM5p886zD3ji1fd, height=0, is_verified=False, tx.id=6623daf81edb66807926b86b4c059eba78e8c2663258c2dc558dd45cb93c3627
2019-10-04 11:17:34,819 INFO     torba.client.baseledger:128: (lbc_mainnet) on_address: ['bRUhAwDH8fv6bbDzkGmnzLtKa2GMGkencA', 'bHRusgRpZhEpTkkPdPYKcLS5KeJ8To3bEB', 'bXXw1aUd5dn548zNVTAsg4287sbGATT5YW', 'bSjjaQAuqoJtHgvhxhNDa1JNKxvE9hkJZ2', 'bQaDzUv6kUuYkQxFBZauEgAJXPvohs4ixi']
Task exception was never retrieved
future: <Task finished coro=<BaseLedger.update_history() done, defined at c:\users\solodolo\lbry\torba\torba\client\baseledger.py:411> exception=ValueError('Transaction 2204aa219b43e5ba9f05b2772da995f78f0960c2bae0502bc821e132c2f82251 was not in database and not on network.')>
Traceback (most recent call last):
  File "c:\users\solodolo\lbry\torba\torba\client\baseledger.py", line 441, in update_history
    tx = await task
  File "c:\users\solodolo\lbry\torba\torba\client\baseledger.py", line 525, in cache_transaction
    raise ValueError(f'Transaction {txid} was not in database and not on network.')
ValueError: Transaction 2204aa219b43e5ba9f05b2772da995f78f0960c2bae0502bc821e132c2f82251 was not in database and not on network.
2019-10-04 11:18:14,964 INFO     torba.client.baseledger:121: (lbc_mainnet) on_transaction: address=bHaRgUgEpRQq2HJp4yxVK2ZU8Rqdq9nCax, height=-1, is_verified=False, tx.id=2204aa219b43e5ba9f05b2772da995f78f0960c2bae0502bc821e132c2f82251
2019-10-04 11:18:14,967 INFO     torba.client.baseledger:121: (lbc_mainnet) on_transaction: address=bareVcfVbXEj6Zd2uZ9qLSXtFMFKrXQhLE, height=-1, is_verified=False, tx.id=2204aa219b43e5ba9f05b2772da995f78f0960c2bae0502bc821e132c2f82251
2019-10-04 11:18:15,020 INFO     torba.client.baseledger:128: (lbc_mainnet) on_address: ['bV3rW4eQEM8YPLGNvcS3KyteF6ugmsZFm3', 'bZe1Uxhq2BRGxXDVHAHEfrMhPbRBWnMesc']```

**Comment by tzarebczan** [comment]
Still getting it and other lbry.tv users are seeing long support create times as well. 

```
2019-11-21 21:52:38,466 INFO     torba.client.baseledger:121: (lbc_mainnet) on_transaction: address=bMLK8DAiAzNQ36tXhRy91ECUY4DH4a8XvY, height=0, is_verified=False, tx.id=a7889feff4faac3a71d73309a213d5cb23328fe7eb61934426c0ab8f3d1ef967
2019-11-21 21:52:38,472 INFO     torba.client.baseledger:121: (lbc_mainnet) on_transaction: address=bLREx4CnWCgyjrkwVq6djdYjMSYyg1fMDs, height=0, is_verified=False, tx.id=a7889feff4faac3a71d73309a213d5cb23328fe7eb61934426c0ab8f3d1ef967
2019-11-21 21:52:38,474 INFO     torba.client.baseledger:121: (lbc_mainnet) on_transaction: address=bWyahc6oQBxgXK8jDxPSCnxPR7o9hZdiLG, height=0, is_verified=False, tx.id=a7889feff4faac3a71d73309a213d5cb23328fe7eb61934426c0ab8f3d1ef967
Task exception was never retrieved
future: <Task finished coro=<BaseLedger.update_history() done, defined at torba\client\baseledger.py:419> exception=ValueError('Transaction fb234bd5084832559e11c92a9d3b368e10b912aa129c1c962a48a6d63cd46c4d was not in database and not on network.')>
Traceback (most recent call last):
  File "torba\client\baseledger.py", line 449, in update_history
  File "torba\client\baseledger.py", line 533, in cache_transaction
ValueError: Transaction fb234bd5084832559e11c92a9d3b368e10b912aa129c1c962a48a6d63cd46c4d was not in database and not on network.
Task exception was never retrieved
future: <Task finished coro=<BaseLedger.update_history() done, defined at torba\client\baseledger.py:419> exception=ValueError('Transaction fb234bd5084832559e11c92a9d3b368e10b912aa129c1c962a48a6d63cd46c4d was not in database and not on network.')>
Traceback (most recent call last):
  File "torba\client\baseledger.py", line 449, in update_history
  File "torba\client\baseledger.py", line 533, in cache_transaction
ValueError: Transaction fb234bd5084832559e11c92a9d3b368e10b912aa129c1c962a48a6d63cd46c4d was not in database and not on network.
Task exception was never retrieved
future: <Task finished coro=<BaseLedger.update_history() done, defined at torba\client\baseledger.py:419> exception=ValueError('Transaction fb234bd5084832559e11c92a9d3b368e10b912aa129c1c962a48a6d63cd46c4d was not in database and not on network.')>
Traceback (most recent call last):
  File "torba\client\baseledger.py", line 449, in update_history
  File "torba\client\baseledger.py", line 533, in cache_transaction
ValueError: Transaction fb234bd5084832559e11c92a9d3b368e10b912aa129c1c962a48a6d63cd46c4d was not in database and not on network.
```

**Comment by tzarebczan** [comment]
This was fixed, #2639 is the remaining issue
